pymeasure/instruments/philips/PM6669.py

```diff
@@ -0,0 +1,202 @@
+# THE SOFTWARE.
+#
+
+from datetime import datetime
+from enum import Enum, IntFlag
+from queue import Queue
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+class Functions(Enum):
+    FREQUENCY_A = 1
+    FREQUENCY_B = 2
+    RPM_A = 4
+    PER_A = 8
+    WIDTH_A = 16
+    PWIDTH_A = 32
+    TOT_A = 64
+
+
+class HardwareErrorException(Exception):
+    pass
+
+
+class SpollStatus(IntFlag):
+    """IntFlag type that represents the status of the device."""
+
+    MEASUREMENT_READY = 1
+    READY_FOR_TRIGGERING = 2
+    MEASURING_START_ENABLE = 4
+    MEASURING_STOP_ENABLE = 8
+    GATE_OPEN = 16
+    ERROR = 32
+    SRQ = 64
+    UNUSED2 = 128
+
+
+class MSRFlag(IntFlag):
+    """IntFlag type to build the mask that triggers an service request (SRQ).
+
+    Set this via the MSR command.
+    """
+
+    MEASUREMENT_READY = 1
+    READY_FOR_TRIGGERING = 2
+    MEASURING_START_ENABLE = 4
+    MEASURING_STOP_ENABLE = 8
+    PROGRAMMING_ERROR = 16
+    HARDWARE_FAULT = 32
+    TIME_OUT = 64
+    UNUSED = 128
+
+
+class PM6669(Instrument):
+    """Represents the Philips PM 6669 instrument."""
+
+    def __init__(self, resourceName, **kwargs):
+        super().__init__(
+            resourceName,
+            "Philips PM 6669",
+            **kwargs
+        )
+        self.write("EOI ON")
+        self.freerun = False
+        self.backlog = Queue()
+
+    KEYWORDS = ["FRE", "PER", "WID", "RPM", "PWI"]
+
+    def spoll(self):
+        """Read the status of the device."""
+        try:
+            return SpollStatus(int(self.ask("++spoll")))
+        except TypeError:
+            return SpollStatus(0)
+
+    def trigger(self):
+        """Trigger the device when not in freerun mode."""
+        self.write("X")
+
+    def read_measurement(self):
+        """Wait for an SRQ from the device and then reads the result.
+
+        Require MSR to be set to MSRFlag.MEASUREMENT_READY
+        """
+        self.adapter.wait_for_srq(delay=0)
+        reply = super().read()
+        if reply[:3] in self.KEYWORDS:
+            return float(reply.split(" ")[-1])
+        else:
+            return None
+
+    def ask(self, s):
+        """Override ask method to use our own read function."""
+        self.write(s)
+        return self.read()
+
+    def read(self):
+        """Read function with instrument bug work around.
+
+        If a request is made to the device while a measurement is ready, both the reply and the
+        measurement are returned. The measurement is filtered out and put in a backlog Queue.
+        """
+        reply = ''
+        result = None
+        while reply == '':
+            reply = super().read()
+            reply = reply.strip('\x00')
+            for line in reply.splitlines():
+                if line[:3] in self.KEYWORDS:
+                    self.backlog.put([datetime.now(), line])
```

Comment:
Where is the backlog emptied? I just see a put, this would endlessly accumulate measurements? At least define a method that empties the buffer and returns its contents as appropriate.